### DISCUSSION TITLE
Apple MLX for native mac models

### DISCUSSION THREAD

**Discussion by gmaijoe** [open-webui/open-webui#191] [Ideas]
what would be needed to integrate MLX (https://github.com/ml-explore/mlx) for ollama? I'm running this on a m1 max right now, and anticipate MLX being an important part of running inference at the edge open source

**Comment by krsmes** [comment]
just curious, would this not be more of a suggestion for ollama than for ollama-webui?

**Reply by justinh-rahb** [reply]
Indeed, and maybe not even them since they're currently very tied to llama.cpp as the inference engine. Not sure how MLX would fit into llama.cpp since it already has Metal support, and it's main purpose is running quantized models. I may be wrong but the main feature of MLX right now is being able to run unquantized models on Metal. That's great if you've got a lot of RAM and don't mind things being slower, but not sure it's very useful to the Ollama project.
